app/jobs/patient_nhs_number_lookup_with_pending_changes_job.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class PatientNHSNumberLookupWithPendingChangesJob < ApplicationJob
+  include NHSAPIConcurrencyConcern
+
+  queue_as :imports
+
+  def perform(patient)
+    patient_with_pending_changes = patient.with_pending_changes
+    patient_with_pending_changes.nhs_number =
+      nil unless patient.nhs_number_changed?
+
+    return unless patient_with_pending_changes.changed?
+
+    if patient_with_pending_changes.nhs_number.present? &&
+         !patient_with_pending_changes.invalidated?
+      return
+    end
```

Comment:
The goal of this bit is really unclear to me. Is it just me? Would it be worth adding a comment just to make it a bit clearer what it is doing? Or maybe put it behind a named method?

Reply:
Yeah agreed. It's basically [doing this logic](https://github.com/nhsuk/manage-vaccinations-in-schools/blob/6c3ffb95cb7fe63d4b4d8bf5010c01496f828920/app/jobs/patient_nhs_number_lookup_job.rb#L9) but for the patient with pending changes, but I will make it clearer.